lib/Runtime/Library/JavascriptObject.cpp

```diff
@@ -357,7 +357,127 @@ namespace Js
         return ToStringHelper(args[0], scriptContext);
     }
 
-    JavascriptString* JavascriptObject::ToStringTagHelper(Var thisArg, ScriptContext* scriptContext, TypeId type)
+    // ES2017 19.1.3.6 Object.prototype.toString()
+    JavascriptString* JavascriptObject::ToStringTagHelper(Var thisArg, ScriptContext *scriptContext, TypeId type)
+    {
+        JavascriptLibrary *library = scriptContext->GetLibrary();
+
+        // 1. If the this value is undefined, return "[object Undefined]".
+        if (type == TypeIds_Undefined)
+        {
+            return library->CreateStringFromCppLiteral(_u("[object Undefined]"));
+        }
+        // 2. If the this value is null, return "[object Null]".
+        if (type == TypeIds_Null)
+        {
+            return library->CreateStringFromCppLiteral(_u("[object Null]"));
+        }
+
+        // 3. Let O be ToObject(this value).
+        RecyclableObject *thisArgAsObject = RecyclableObject::FromVar(JavascriptOperators::ToObject(thisArg, scriptContext));
+
+        // 4. Let isArray be ? IsArray(O).
+        // There is an implicit check for a null proxy handler in IsArray, so use the operator.
+        BOOL isArray = JavascriptOperators::IsArray(thisArgAsObject);
+
+        Var tag = nullptr;
+        // 15. Let tag be ? Get(O, @@toStringTag).
+        try
+        {
+            tag = JavascriptOperators::GetProperty(thisArgAsObject, PropertyIds::_symbolToStringTag, scriptContext); // Let tag be the result of Get(O, @@toStringTag).
```

Comment:
question mark in the 15 clause says ReturnIfAbrupt. So if the GetProperty throws an exception shouldn't we let it go instead of handling it here?